packages/env/lib/init-config.js

```diff
@@ -280,36 +247,50 @@ USER root`;
 }
 
 /**
- * Checks the configured PHP version
- * against the minimum version supported by Xdebug
+ * Gets the Xdebug config based on the options in the config object.
  *
- * @param {WPConfig} config
- * @return {boolean} Whether the PHP version is supported by Xdebug
+ * @param {string} xdebugMode The Xdebug mode set in the config.
+ * @param {string} phpVersion The php version set in the environment.
+ * @return {string} The Xdebug config -- can be an empty string when it's not used.
  */
-function checkXdebugPhpCompatibility( config ) {
-	// By default, an undefined phpVersion uses the version on the docker image,
-	// which is supported by Xdebug 3.
-	const phpCompatibility = true;
-
-	// If PHP version is defined
-	// ensure it meets the Xdebug minimum compatibility requirment.
-	if ( config.env.development.phpVersion ) {
-		const versionTokens = config.env.development.phpVersion.split( '.' );
+function getXdebugConfig( xdebugMode = 'off', phpVersion ) {
+	if ( xdebugMode === 'off' ) {
+		return '';
+	}
+
+	let xdebugVersion = 'xdebug';
+
+	if ( phpVersion ) {
+		const versionTokens = phpVersion.split( '.' );
 		const majorVer = parseInt( versionTokens[ 0 ] );
 		const minorVer = parseInt( versionTokens[ 1 ] );
 
 		if ( isNaN( majorVer ) || isNaN( minorVer ) ) {
-			throw new Error(
+			throw new ValidationError(
 				'Something went wrong when parsing the PHP version.'
 			);
 		}
 
-		// Xdebug 3 supports 7.2 and higher
-		// Ensure user has specified a compatible PHP version.
+		// Throw an error if someone tries to use Xdebug with an unsupported PHP version.
+		// Xdebug 3 only supports 7.2 and higher.
 		if ( majorVer < 7 || ( majorVer === 7 && minorVer < 2 ) ) {
-			throw new Error( 'Cannot use XDebug 3 on PHP < 7.2.' );
+			throw new ValidationError(
+				`Cannot use XDebug 3 with PHP < 7.2. Your PHP version is ${ phpVersion }.`
+			);
+		}
+
+		// For now, we support PHP 7 by installing the final version of Xdebug to
+		// support PHP 7 when the environment uses that version. By default, use the
+		// latest version.
+		if ( majorVer === 7 ) {
+			xdebugVersion = 'xdebug-3.1.6';
```

Comment:
What do you think about using the [support version table from Xdebug's website](https://xdebug.org/docs/compat)? We can pair each PHP minor release with a compatible version of Xdebug and then fall back to plain `xdebug` to grab the latest.

Reply:
Something like:

```
// Map the PHP version to a compatible Xdebug version.
const versionMap = {
	'7.2': 'xdebug-3.1.6',
	'7.3': 'xdebug-3.1.6',
	'7.4': 'xdebug-3.1.6',
	'8.0': 'xdebug-3.2.1',
};
const xdebugVersion = versionMap[ `${ majorVer }.${ minorVer }` ] ?? 'xdebug';
```

Reply:
I think it might be useful in the future, but I don't know if it provides any value right now -- we either use current (3.2), or previous (3.1). Outside of that, we'd be on 2.x, which would require config changes that would be hard to maintain